tests/test_slurm_system.py

```diff
@@ -39,24 +55,73 @@ def test_parse_squeue_output_with_node_ranges_and_root_user(slurm_system):
 
 
 def test_parse_sinfo_output(slurm_system):
-    sinfo_output = (
-        "PARTITION AVAIL TIMELIMIT NODES STATE NODELIST\n"
-        "main up infinite 1 idle nodeA001\n"
-        "main up infinite 1 idle nodeB001"
-    )
-    node_user_map = {"nodeA001": "root", "nodeB001": "user"}
+    sinfo_output = """PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
+    main    up    3:00:00      1  inval node-081
+    main    up    3:00:00      5  drain node-[065-066,114,124-125]
+    main    up    3:00:00      2   resv node-[034-035]
+    main    up    3:00:00     88  alloc node-[033,036-064,067-080,082-113,115-123,126-128]
+    backup    up   12:00:00     16  idle node[01-16]
+    """
+    node_user_map = {
+        "": "user1",
+        "node-033": "user2",
+        "node-034": "user3",
+        "node-056": "user3",
+        "node-057": "user3",
+        "node-058": "user3",
+        "node-049": "user4",
+        "node-050": "user4",
+        "node01": "user5",
+        "node02": "user5",
+        "node03": "user5",
+        "node04": "user5",
+        "node05": "user5",
+        "node06": "user5",
+        "node07": "user5",
+        "node08": "user5",
+    }
     slurm_system.parse_sinfo_output(sinfo_output, node_user_map)
-    assert slurm_system.partitions["main"][0].state == SlurmNodeState.IDLE
-    assert slurm_system.partitions["main"][1].state == SlurmNodeState.IDLE
+    for i in range(len(slurm_system.partitions["main"])):
+        assert slurm_system.partitions["main"][i].state == SlurmNodeState.ALLOCATED
```

Comment:
According to your `sinfo_output`, nodes have different states, but after parsing all of them are allocated. Is that expected?

Reply:
according to sinfo_output, all nodes in main are allocated and all nodes in backup are idle so the behaviour seems good for me

Reply:
Are we talking about the same code? :)

```py
    sinfo_output = """PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
    main    up    3:00:00      1  inval node-081                      # <- NOT ALLOCATED
    main    up    3:00:00      5  drain node-[065-066,114,124-125]    # <- NOT ALLOCATED
    main    up    3:00:00      2   resv node-[034-035]                # <- NOT ALLOCATED
    main    up    3:00:00     88  alloc node-[033,036-064,067-080,082-113,115-123,126-128]
    backup    up   12:00:00     16  idle node[01-16]
    """
```

Reply:
ok I see the misunderstanding
Indeed not all nodes are allocated in the example but if you look at the slurm system, we're just checking nodes 115 to 122 (in main) which are indeed allocated

Reply:
updated